jlm/hls/ir/hls.cpp

```diff
@@ -4,15 +4,40 @@
  */
 
 #include <jlm/hls/ir/hls.hpp>
+#include <jlm/util/HashSet.hpp>
 
 namespace jlm::hls
 {
 
+std::size_t
+triggertype::ComputeHash() const noexcept
+{
+  return std::hash<const triggertype *>()(GetInstance());
```

Comment:
I would suggest not hashing on some address, but simply using a constant magic value

Something like this would also work (it gets compile-time evaluated):

`constexpr std::size_t const_hash_str(std::string_view s) {
	std::size_t v = 0;
	for (auto c : s) {
		v = v * 5 + c;
	}
	return v;
}

std::size_t triggertype::ComputeHash() const noexcept { return constr_str_hash("triggertype"); }
`

Reply:
I am fine with either, but why not hashing on a constant address? What is the rationale?

Reply:
a) determinism b) compile-time constants are always nice I would say

Reply:
Computing the hash at compile-time now where possible. I did not find a way to automatically get the qualified name of the type (`typeid(triggertype).name` did not do the trick), so I sadly had to provide manually the string.